tests/moses/main/ASMOSESUTest.cxxtest

```diff
@@ -0,0 +1,128 @@
+/** ASMOSESUTest.cxxtest ---
+ *
+ * Copyright (C) 2018 OpenCog Foundation
+ *
+ * Author: Behailu Dereje <Behailu04@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <cxxtest/TestSuite.h>
+#include <opencog/moses/main/moses_exec.h>
+#include <opencog/util/Logger.h>
+#include <boost/current_function.hpp>
+#include <opencog/moses/scoring/bscores.h>
+#include <examples/example-progs/scoring_iterators.h>
+#include <opencog/atoms/base/Link.h>
+#include "moses-framework.h"
+
+using namespace opencog;
+using namespace combo;
+using namespace moses;
+using namespace std;
+
+const string MOSESUTest_dir = string(PROJECT_SOURCE_DIR) + "/tests/moses/main/";
+
+/**
+ * MOSES test on toy problems
+ */
+class ASMOSESUTest : public CxxTest::TestSuite
+{
+public:
+    ASMOSESUTest() {
```

Comment:
@behailu04, since it is a newly created file, could you indent it with smart tabs?